### ISSUE TITLE
[networkd] DHCP4 CLIENT: Failed to set request flag for static route: File exists

### ISSUE THREAD

**Issue by chr0mag** [systemd/systemd#16964]
**systemd version the issue has been seen with**
> % systemctl --version
systemd 246 (246.4-1-arch)
+PAM +AUDIT -SELINUX -IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +ZSTD +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid

**Used distribution**
 > ArchLinux

**Expected behaviour you didn't see**
 > Wireless network is connected on boot.

**Unexpected behaviour you saw**
 > Wireless network fails to connect on boot. Restarting systemd-networkd manually after boot works.

**Steps to reproduce the problem**
I have the following wireless card.
```
04:00.0 Network controller: Intel Corporation Wireless 8260 (rev 3a)
	Subsystem: Intel Corporation Device 1130
	Kernel driver in use: iwlwifi
	Kernel modules: iwlwifi
```
Configured using the following .link and .network files.
_/etc/systemd/network/20-wifi0.link_ 
```
[Match]
MACAddress=...

[Link]
Description=wifi0 link
Name=wifi0
```
_/etc/systemd/network/20-wifi0-dhcp.network_ 
```
[Match]
Name=wifi0

[Network]
DHCP=ipv4

[DHCPv4]
RouteMetric=20
```
This fails to acquire an IP on every system boot and the error in the title appears in the logs. This particular error began appearing after upgrading from 245.7-1 -> 246-1. Prior to that I would see periodic failures to acquire an IP on boot and the following error would appear in the logs: `wifi0: DHCP4 CLIENT: Failed to attach event: Device or resource busy` . This 2nd error began after upgrading from 244.3-1 -> 245-3.

Restarting networkd manually after boot is complete works every time. 

Related issues:
* https://bugs.archlinux.org/task/66433
* https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/1870410

The Ubuntu bug report in particular contains a [patch](https://git.launchpad.net/~ubuntu-core-dev/ubuntu/+source/systemd/commit/?id=4c48893eb04b01a2ec62d2d2823a79a9f5cb2b80) that may be useful.

I've attached the journal output from a boot with debug enabled for systemd-networkd.
[networkd-debug.log](https://github.com/systemd/systemd/files/5178940/networkd-debug.log)

**Comment by chr0mag** [comment]
'...every system boot...' 
Not quite.
```
% COUNT=0 && for i in {0..83}; do if ! journalctl --boot=-${i} --unit=systemd-networkd.service --no-pager | grep -i 'dhcp4 client' ; then ((++COUNT)) ; fi ; done ; echo $COUNT
Oct 03 18:22:00 thinkpad systemd-networkd[584]: wifi0: DHCP4 CLIENT: Failed to set request flag for static route: File exists
...
Aug 13 21:02:55 thinkpad systemd-networkd[608]: wifi0: DHCP4 CLIENT: Failed to set request flag for static route: File exists
9
```
So 9 times out of the last 83 boot attempts a dhcp4 IP was acquired on boot without having to restart networkd.

**Comment by yuwata** [comment]
The issue `wifi0: DHCP4 CLIENT: Failed to attach event: Device or resource busy `should be fixed by 571eeba9092e18b9f082f8c933ecc1ae20e8997f.

**Comment by yuwata** [comment]
Fix is waiting in #17341.

@chr0mag If possible, please test the PR.
